### DISCUSSION TITLE
Write a big file, and cancel before flush, when will the uploaded chunk file deleted?

### DISCUSSION THREAD

**Discussion by ningfdx** [seaweedfs/seaweedfs#3142] [Q&A] [answered]
SeaweedFS mount client store chunk info in it's memory, if I cancel a big file write flow before do flush, the chunk files in volume that uploaded before will never deleted. Is it true?

I found seaweedfs/weed/mount/weedfs_file_write.go:35 Write() ignore the cancle signal, and filer does't store [file->chunk list] info, so I have this question.

![image](https://user-images.githubusercontent.com/17705801/172151469-48eb2bfb-3e68-4551-9c28-db84dbe5e459.png)

ps: mount process be killed will cause the same situation. Is it possible to tell filer to do this, or set a big ttl like 24Hours to recycle automatically

**Comment by chrislusf** [comment, marked as answer]
This is possible but should be rare. Current code handles the cancel in the middle. What could happen is when the server dies suddenly, or one of the copy fail to write and the delete operation also fails.

To garbage collect the dangling fileIds, the `volume.fsck` in the `weed shell` can help. It is a computation intensive process though. Maybe run it when the system is not so busy.

**Reply by ningfdx** [reply]
OK, good design! I understand it now, Thanks for your reply.
